### PR TITLE
[FLINK-10208][build] Bump mockito to 2.21.0 / powermock to 2.0.0-beta.5

### PR DESCRIPTION
**Pull request by zentol**
## What is the purpose of the change

This PR bumps mockito to 2.21.0 and powermock to 2.0.0-beta.5 for jdk9 compatibility.

This is a substantial version bump, as our current mockito version is 10 years old.

As a result a fair number of incompatibilities had to be resolved. I grouped instances of the same issue into single commits, and added a note to each commit as to what has changed.

Some of these are rather mechanical (like package changes), while others are pretty subtle (like the exception matching).

### CODE REVIEW

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/BarrierBufferTestBase.java [hunk 1]
```diff
@@ -531,7 +532,7 @@ public void testMultiChannelSkippingCheckpoints() throws Exception {
 
 		// checkpoint 3 aborted (end of partition)
 		check(sequence[20], buffer.getNextNonBlocked(), PAGE_SIZE);
-		verify(toNotify).abortCheckpointOnBarrier(eq(3L), any(CheckpointDeclineSubsumedException.class));
+		verify(toNotify).abortCheckpointOnBarrier(eq(3L), any(InputEndOfStreamException.class));
```
Comment (zentol) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/BarrierBufferTestBase.java]: @NicoK @zhijiangW I'd love to get your input here. As far as i can tell the existing code was simple wrong, as the supposed exception wasn't supposed to be thrown in this situation.

File: pom.xml [hunk 2]
```diff
@@ -171,7 +171,7 @@ under the License.
 
 		<dependency>
 			<groupId>org.mockito</groupId>
-			<artifactId>mockito-all</artifactId>
```
Comment (zentol) [inline pom.xml]: The mockito-all artifact is no longer distributed.

File: flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/slot/TimerServiceTest.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 import org.apache.flink.util.TestLogger;
 
 import org.junit.Test;
-import org.mockito.internal.util.reflection.Whitebox;
+import org.powermock.reflect.Whitebox;
```
[the remark this answers is not in the document]
Comment (zentol) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/slot/TimerServiceTest.java]: We could, but in the long term I'd like to get rid of powermockito directly.

[the remark this answers is not in the document]
Comment (azagrebin) [reply flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/slot/TimerServiceTest.java]: ok, does it make sense then to use everywhere one thing? e.g. then `flink.Whitebox`?

File: flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendSavepointTest.java [hunk 4]
```diff
@@ -315,7 +315,7 @@ private static void restoreStdOutAndStdErr() {
 	private static ClusterClient<String> createClusterClient(String expectedResponse) throws Exception {
 		final ClusterClient<String> clusterClient = mock(ClusterClient.class);
 
-		when(clusterClient.triggerSavepoint(any(JobID.class), anyString()))
+		when(clusterClient.triggerSavepoint(any(JobID.class), nullable(String.class)))
```
Comment (TisonKun) [inline flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendSavepointTest.java]: @zentol What is the preference between `nullable(<any>)` and `any(<class>)`? Inconsistent on this, take a look at `HadoopOutputFormatTest.java` where change `any(FileSystem.class)` to `nullable(FileSystem.class)`

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/BarrierBufferTestBase.java (hunk 1 above)
Comment (zentol) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/BarrierBufferTestBase.java]: alright so we need a matcher that actually checks the class?

Comment (zentol) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/BarrierBufferTestBase.java]: I supposed we should use `isA` instead.

File: pom.xml [hunk 5]
```diff
@@ -123,8 +123,8 @@ under the License.
 		<prometheus.version>0.3.0</prometheus.version>
 		<avro.version>1.8.2</avro.version>
 		<junit.version>4.12</junit.version>
-		<mockito.version>1.10.19</mockito.version>
-		<powermock.version>1.6.5</powermock.version>
+		<mockito.version>2.21.0</mockito.version>
+		<powermock.version>2.0.0-beta.5</powermock.version>
```
[the remark this answers is not in the document]
Comment (zentol) [reply pom.xml]: There's no proper 2.0.0 release yet :/ This is the only version available on maven central, but it's already used a fair bit.

https://mvnrepository.com/artifact/org.powermock/powermock-core

**Pull request merged**